### DISCUSSION TITLE
`deleteMany` fetches records first?

### DISCUSSION THREAD

**Discussion by baptisteArno** [prisma/prisma#17949] [Q&A] [answered]
When I try to:

```ts
const resultIds = ["id1", "id2"]

prisma.answer.deleteMany({
        where: {
          resultId: { in: resultIds },
        },
      })
```

When the model for `answer` is:

```
model Answer {
  resultId    String
  blockId     String
  groupId     String
  content     String   @db.Text
  result      Result   @relation(fields: [resultId], references: [id], onDelete: Cascade)

  @@unique([resultId, blockId, groupId])
}
```

It will first fetch all the answer records by unique indexes, then delete them all.

Is there a way to prevent this? As I want it to directly use the resultIds list. And this query is critical in terms of performance.

**Comment by nurul3101** [comment, marked as answer]
Hello @baptisteArno 👋 

It's not possible to change the query behaviour at the moment, you would need to use [Raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) in case you need to delete the records in a single query.

Here's a similar GitHub Discussion regarding the same topic
- https://github.com/prisma/prisma/discussions/8211

And the feature request
- https://github.com/prisma/prisma/issues/8239

**Reply by baptisteArno** [reply]
Ok thanks, sorry for the duplication, I struggled to search for similar requests 🙏

**Reply by nurul3101** [reply]
No worries at all! :) 
Sometimes duplication helps, some other users might benefit from this GitHub Discussion as well! 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
